detection_rules/rule.py

```diff
@@ -3,294 +3,290 @@
-
-    def validate(self, as_rule=False, versioned=False, query=True):
-        """Validate against a rule schema, query schema, and linting."""
-        self.normalize()
-
-        if as_rule:
-            schema_cls = CurrentSchema.toml_schema()
-            contents = self.rule_format()
-        elif versioned:
-            schema_cls = CurrentSchema.versioned()
-            contents = self.contents
-        else:
-            schema_cls = CurrentSchema
-            contents = self.contents
+@dataclass(frozen=True)
+class LuceneRuleData(BaseQueryRuleData):
+    """Specific fields for query event types."""
+    language: Literal["lucene"]
 
-        schema_cls.validate(contents, role=self.type)
 
-        skip_query_validation = self.metadata['maturity'] in ('experimental', 'development') and \
-            self.metadata.get('query_schema_validation') is False
+@dataclass(frozen=True)
+class MachineLearningRuleData(BaseRuleData):
+    type: Literal["machine_learning"]
 
-        if query and self.query is not None and not skip_query_validation:
-            ecs_versions = self.metadata.get('ecs_version', [ecs.get_max_version()])
-            beats_version = self.metadata.get('beats_version', beats.get_max_version())
-            indexes = self.contents.get("index", [])
+    anomaly_threshold: int
+    machine_learning_job_id: str
 
-            if self.contents['language'] == 'kuery':
-                self._validate_kql(ecs_versions, beats_version, indexes, self.query, self.name)
 
-            if self.contents['language'] == 'eql':
-                self._validate_eql(ecs_versions, beats_version, indexes, self.query, self.name)
+@dataclass(frozen=True)
+class ThresholdQueryRuleData(BaseQueryRuleData):
+    """Specific fields for query event types."""
 
-    @staticmethod
-    @cached
-    def _validate_eql(ecs_versions, beats_version, indexes, query, name):
-        # validate against all specified schemas or the latest if none specified
+    @dataclass(frozen=True)
+    class ThresholdMapping:
+        @dataclass(frozen=True)
+        class ThresholdCardinality:
+            field: str
+            value: definitions.ThresholdValue
+
+        field: List[str]
+        value: definitions.ThresholdValue
+        cardinality: Optional[ThresholdCardinality]
+
+    type: Literal["threshold"]
+    language: Literal["kuery", "lucene"]
+    threshold: ThresholdMapping
+
+
+@dataclass(frozen=True)
+class EQLRuleData(BaseQueryRuleData):
+    """EQL rules are a special case of query rules."""
+    type: Literal["eql"]
+
+    @property
+    def parsed_query(self) -> kql.ast.Expression:
+        with eql.parser.elasticsearch_syntax, eql.parser.ignore_missing_functions:
+            return eql.parse_query(self.query)
+
+    @property
+    def unique_fields(self):
+        return list(set(str(f) for f in self.parsed_query if isinstance(f, eql.ast.Field)))
+
+    def validate_query(self, beats_version: str, ecs_versions: List[str]):
```

Comment:
all validation (schema, query, ecs, beats, etc.), will need to be bypassed for all rules which are `maturity = "deprecated"`. Ref: #1050

Reply:
Already been implemented
https://github.com/rw-access/detection-rules/blob/400cb0673be71f6dd3c3baccbfcd82eb8a7c5dad/detection_rules/rule.py#L365-L367

Reply:
I can add `maturity = deprecated` to that

Reply:
c66b8eaaa1ab5ea324b0089d24d1e7613c6c0941